neuraxle/data_container.py

```diff
@@ -0,0 +1,242 @@
+
+..
+    Copyright 2019, Neuraxio Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+..
+
+    Thanks to Umaneo Technologies Inc. for their contributions to this Machine Learning
+    project, visit https://www.umaneo.com/ for more information on Umaneo Technologies Inc.
+
+"""
+import hashlib
+from typing import Any, Iterable, List
+
+from conv import convolved_1d
+
+from neuraxle.hyperparams.space import HyperparameterSamples
+
+
+class DataContainer:
+    """
+    DataContainer class to store data inputs, expected outputs, and ids together.
+    Each :class:`BaseStep` needs to rehash ids with hyperparameters so that the :class:`Checkpoint` step
+    can create checkpoints for a set of hyperparameters.
+
+    The DataContainer object is passed to all of the :class:`BaseStep` handle methods :
+        * :func:`~neuraxle.base.BaseStep.handle_transform`
+        * :func:`~neuraxle.base.BaseStep.handle_fit_transform`
+        * :func:`~neuraxle.base.BaseStep.handle_fit`
+
+    Most of the time, you won't need to care about the DataContainer because it is the pipeline that manages it.
+
+    .. seealso::
+        :class:`BaseHasher`,
+        :class: `BaseStep`
+    """
+
+    def __init__(self,
+                 current_ids,
+                 data_inputs: Any,
+                 expected_outputs: Any = None
+                 ):
+        self.current_ids = current_ids
+        self.data_inputs = data_inputs
+        if expected_outputs is None:
+            self.expected_outputs = [None] * len(current_ids)
+        else:
+            self.expected_outputs = expected_outputs
+
+    def set_data_inputs(self, data_inputs: Iterable):
+        """
+        Set data inputs.
+
+        :param data_inputs: data inputs
+        :type data_inputs: Iterable
+        :return:
+        """
+        self.data_inputs = data_inputs
+
+    def set_expected_outputs(self, expected_outputs: Iterable):
+        """
+        Set expected outputs.
+
+        :param expected_outputs: expected outputs
+        :type expected_outputs: Iterable
+        :return:
+        """
+        self.expected_outputs = expected_outputs
+
+    def set_current_ids(self, current_ids: List[str]):
+        """
+        Set current ids.
+
+        :param current_ids: data inputs
+        :type current_ids: List[str]
+        :return:
+        """
+        self.current_ids = current_ids
+
+    def summary_hash(self):
```

Comment:
I see you added this method in the data container. cool. you should be able to delete this method from the BaseStep and from the BaseHasher.

Reply:
yes forgot to do it